src/CommitmentsV2/SFA.DAS.CommitmentsV2.Api/Controllers/AuthorizationController.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System.Threading.Tasks;
+using MediatR;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using SFA.DAS.CommitmentsV2.Api.Types.Requests;
+using SFA.DAS.CommitmentsV2.Application.Queries.GetCohortSummary;
+using SFA.DAS.CommitmentsV2.Types;
+
+namespace SFA.DAS.CommitmentsV2.Api.Controllers
+{
+    [Route("api/authorization")]
+    [Authorize]
+    public class AuthorizationController : ControllerBase
+    {
+        private readonly IMediator _mediator;
+
+        public AuthorizationController(
+            IMediator mediator)
+        {
+            _mediator = mediator;
+        }
+
+        [HttpGet]
+        [Route("access-cohort")]
+        public async Task<IActionResult> CanAccessCohort(PartyType partyType, string partyId, long cohortId)
+        {
+            var cohort = await _mediator.Send(new GetCohortSummaryRequest{CohortId = cohortId});
```

Comment:
Hmm I'm not sure piggybacking on `GetCohortSummaryRequest` is the most efficient way to check if the operation is authorized given that the query result is a projection. It's possible that the query will evolve over time too and return additional properties (potentially involving joins etc).

Is there a more efficient way to query the DB to check if the operation is authorized? e.g. A specific query:

```c#
var canAccessCohort = _db.Value.Commitment.Any(c =>
    c.Id == cohortId && (
    partyType == PartyType.Employer && c.EmployerAccountId == accountId ||
    partyType == PartyType.Provider && c.ProviderId == providerId));
```

Reply:
implemented